### ISSUE TITLE
[Short Term] Rarely Cybran Engineers become invincible to Direct Fire Damage That has NO AOE

### ISSUE THREAD

**Issue by Azraeel** [Team-QUIET/QUIET-Community-Edition#253]

**Comment by PurpleMossClump** [comment]
This is really inconsistent. I haven't found a way to reliably reproduce it.

**Comment by loudpatches** [comment]
Looks like inconsistent bones and collision boxes.  cybrans have a couple of bones pretty high up that might be aim bones.  should check out an older version andd compare the t1, t2, etc. engineer to see what's up.

**Comment by PurpleMossClump** [comment]
Reproduceable! 
Cybran engineers' drones collision boxes are stopping projectiles from hitting the engineer.
Once the engineer builds or repairs something, and the drone goes back to the engineer, the drone's collision box overlaps with the engineer's own, and since the drone is invulnerable, the engineer only takes damage from AoEs. The engineer is still able to be damaged by non-AoE damage if the projectile hits above the drone's collision box or if the engineer is building something, since the drone's collision box follows the drone.

**Comment by PurpleMossClump** [comment]
Did this come from modifying all the units density and whatnot?

**Comment by loudpatches** [comment]
https://github.com/Team-QUIET/QUIET-Community-Edition/commit/b5826d56467b9d9e7c00270c9d6f2c8d23e116b0#diff-ec89b4cdad02c3afba45d9e0f9341e045dfe4f55a1e94fd4bb6a5e3231edc542R427  wondering.

**Comment by PurpleMossClump** [comment]
Yeah, I rolled back to 1.47 and the drone's collision box is tiny, so it doesn't interfere with projectiles.

**Comment by loudpatches** [comment]
@Azraeel looks like the build bot is ura0001.  You need to make an exception for this in the blueprint mods.  When it spawns (via EffectUtilities.lua), it's set to take no damage and be unkillable.

```lua
 tunit = CreateUnit('ura0001', army, x + xVec, y + yVec, z + zVec, qx, qy, qz, qw, 'Air' )

            -- Make build bots unkillable
            tunit:SetCanTakeDamage(false)
            tunit:SetCanBeKilled(false)
```

So when you scaled up the unit's collision box (I think), you created a shield for the engineers; one that can't be killed and can intercept hits.

**Comment by Azraeel** [comment]
Should be fixed now thank you a lot guys!
Im on night shift, now so fixing sleep schedule.
